libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -799,20 +799,29 @@ class MapboxRouteLineApi(
         }
     }
 
+    // The purpose of this is to provide the correct masking layer data to be rendered in the view.
+    // Although the primary route line layer data is returned here it shouldn't be rendered by the
+    // view. It is included because the parameter isn't nullable.  If it were the data wouldn't
+    // be included since it's only the masking layers that should be updated. For this reason
+    // the variable to ignore the primary route line data is set to true so the view class will
+    // ignore it.
     private fun provideRouteLegUpdate(
         routeLineOverlayDynamicData: RouteLineDynamicData?,
         consumer: MapboxNavigationConsumer<Expected<RouteLineError, RouteLineUpdateValue>>
     ) {
         ifNonNull(routeLineOverlayDynamicData) { maskingLayerData ->
             getRouteDrawData { expected ->
                 expected.value?.apply {
+                    val updateValue = RouteLineUpdateValue(
+                        this.primaryRouteLineData.dynamicData,
+                        this.alternativeRouteLinesData.map { it.dynamicData },
+                        maskingLayerData
+                    ).also {
+                        it.ignorePrimaryRouteLineData = true
```

Comment:
This can be dangerous:
1. We set this flag to true;
2. We pass `RouteLineUpdateValue` to the client via consumer;
3. They do something in their consumer which results in passing us not the original object but its copy (`toMutableValue/toImmutableValue`);
4. We lose the flag.

It sounds artificial but who knows. I think we should transfer the flag value between mutable/immutable (we can do that and it seems like the only way the customer can modify the object).

Reply:
What you point out is true. However, before beta.3 this data was never returned in this code execution path. The only reason this was introduced was because the masking layers were added in this release and they had to be created and rendered here.  The main route line data is included ONLY because the field is non-nullable.  If the field had been nullable the main route line parameter would have been null.  So even in the possibility you outline the data shouldn't be rendered and wouldn't be if the field had been nullable or could be changed to nullable (without breaking SEMVER).

The flag is introduced solely as a workaround for not being able to exclude the main route line data in a more elegant way and to get this method back to the state it was in prior to beta.3.

Reply:
> The flag is introduced solely as a workaround for not being able to exclude the main route line data in a more elegant way and to get this method back to the state it was in prior to beta.3.

I get that, no objections to that.
What I mean is imagine the situation:
```
    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
        routeLineApi.updateWithRouteProgress(routeProgress) { result ->
            val modifiedResult = result.onValue { it.toMutableValue()/** some changes **/.toImmutableValue() }
            mapboxMap.getStyle()?.apply {
                routeLineView.renderRouteLineUpdate(this, modifiedResult)
            }
        }
    }    
```
This is possible, right? It doesn't have anything to do with beta.3. But with these modifications we lose the flag. Which I think shouldn't be the case. If we add a new property to the model class (even though it's internal), `toMutableValue + toImmutableValue` should keep it.

Reply:
Yes good point, added.